api/tests/src/db/models/factories.py

```diff
@@ -721,6 +723,8 @@ class Meta:
     is_image_workspace_enabled = factory.Faker("boolean")
     is_validation_workspace_enabled = factory.Faker("boolean")
 
+    top_level_agency = factory.Faker("agency")
```

Comment:
I don't think this would work? The `agency` provider is written towards the top of this file, and just returns a string, but this would need to be another instance of the agency class. I think for the factory, I'd probably just leave this unset by default otherwise you'd need to have it refer to itself and run into a recursive loop issue.